pkg/cmd/init.go

```diff
@@ -16,160 +16,163 @@ limitations under the License.
 package cmd
 
 import (
+	"crypto/x509"
+	"fmt"
 	"net"
+	"os"
 	"path/filepath"
+	"time"
 
 	"k8s.io/apiserver/pkg/authentication/user"
 	ctrl "k8s.io/kubernetes/pkg/controlplane"
 
 	"github.com/openshift/microshift/pkg/config"
 	"github.com/openshift/microshift/pkg/util"
 	"github.com/openshift/microshift/pkg/util/cryptomaterial"
+	"github.com/openshift/microshift/pkg/util/cryptomaterial/certchains"
 )
 
-func initAll(cfg *config.MicroshiftConfig) error {
-	// create CA and keys
-	clusterTrustBundlePEM, certChains, err := initCerts(cfg)
+func initCerts(cfg *config.MicroshiftConfig) (*certchains.CertificateChains, error) {
+	certChains, err := certSetup(cfg)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	// create kubeconfig for kube-scheduler, kubelet,controller-manager
-	if err := initKubeconfig(cfg, clusterTrustBundlePEM, certChains); err != nil {
-		return err
+
+	// we cannot just remove the certs dir and regenerate all the certificates
+	// because there are some long-lived certs and CAs that shouldn't be swapped
```

Comment:
This answers a question I had about why we don't just regenerate all the certs every time we start up. :-)

Reply:
This PR consists of many other PRs changes that need to be merged first and so eventually, the commit number will be much smaller and the changes should seem more contained.

Reply:
As for the longer lived certificates, I think good examples are:
1. kube-apiserver serving certificate CAs - if you're connecting from the outside, the CA swap would likely be confusing
2. system:admin client certificate - you don't want to rotate the certificate that allows you to authenticate to the API